### PR TITLE
Add pep257 checker

### PR DESCRIPTION
**Pull request by lsix**
This have been tested on `pythonPackages` and `python3Packages`. Test with `pypyPackages` sill in progress.

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -2452,6 +2452,24 @@ let
...
+
+     src = pkgs.fetchurl {
+       url = "https://pypi.python.org/packages/source/p/pytest-pep257/${name}.tar.gz";
+       sha256 = "003vdkxpx37n0kjqpwgj3314hwk2jfz3nz58db7xh68bf8xy75lk";
+     };
+
+     propagatedBuildInputs = with self ; [ pytest pep257 ];
+
+     meta = {
+       website = "https://github.com/anderslime/pytest-pep257";
```
Comment (pSub) [inline pkgs/top-level/python-packages.nix]: That field is called `homepage` and you do not need quotes around urls.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -2452,6 +2452,24 @@ let
...
+       description = "py.test plugin for pep257";
```
Comment (pSub) [inline pkgs/top-level/python-packages.nix]: You should not repeat the package name in the description.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -8765,6 +8783,23 @@ let
...
+    name = "pep257-${version}";
+    version = "0.3.2";
+
+    src = pkgs.fetchurl {
+      url = "https://github.com/GreenSteam/pep257/archive/${version}.tar.gz";
+      sha256 = "0v8aq0xzsa7clazszxl42904c3jpq69lg8a5hg754bqcqf72hfrn";
+    };
+
+    meta = {
+      homepage = "https://github.com/GreenSteam/pep257/";
```
Comment (pSub) [inline pkgs/top-level/python-packages.nix]: Quotes again.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (lancelotsix) [inline pkgs/top-level/python-packages.nix]: `py.test` is the name of the testing framework this plugs in (`pythonPackages.pytest` in `nixpkgs`), and `PEP257` the name of the standard this plug-in checks code against. The description (I actually picked from the project homepage) only refers to the same elements the package name is built on.

**Comment by lancelotsix** [comment]
I updated the PR according to the comments.
Thanks.

**Comment by lethalman** [comment]
Built locally thanks.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 8700 [1](https://github.com/Mic92/nixpkgs-review)

**Comment by jonringer** [comment]
oops, mistyped a PR number :)

**Pull request merged**